homeassistant/components/roon/media_player.py

```diff
@@ -65,6 +69,12 @@ async def async_setup_entry(
         "async_transfer",
     )
 
+    platform.async_register_entity_service(
```

Comment:
I believe that the HA Media Player doesn't support the Roon concept of 'Radio', thus may not be able to be neatly passed using the play media service.  If you know of a neater way for this however, it would be greatly appreciated.

The intent is to be able to start a Roon radio from a given artist or genre which is currently not possible.

The RoonAPI Play_Media function is: play_media(self, zone_or_output_id, path, action=None, report_error=True)
Where action can be: Null (for the default play action), or "Play Now", "Queue" or "Start Radio".

The code here for its play_media doesn't pass the 'action' parameter to the roonApi play_media function.
In addition, there is no way to request the 'Start Radio' action through the 'path'.

A separate service for 'Start Radio' would allow the usage of this Roon concept.

Reply:
@balloob is there a way to pass an extra parameter to `play_media`?

There was a previous proposal to reuse the `media_type` parameter for this which I didn’t think you guys would be comfortable with.

Reply:
The Home Assistant Play Media API can process any incoming data. So you could have a URL format like `roon-radio://chill-mix?artist=Donnie`.

You can then expose these items via the media browser so users don't actually have to craft these URLs themselves.

Reply:
So given @balloob's comments suggest you go back to your previous PR - and we find a way to delimit the action at the start or end of the current path string. 

One option would be to enhance the pyroon `split_path`  function to do this.

Reply:
Hi, I’ve pulled together notes for an architecture discussion as a starting point, attempting to summarise the options suggested thus far and potential pros/cons with each.  Appreciate the support and assistance towards implementing this functionality that will be a really powerful enhancement selection of music to play through HA automations.  Let me know if there’s somewhere better to place this content for appropriate collaboration.
 
The purpose for this architecture discussion is to collaboratively identify the best method to enhance the HA play_media service to support requesting music to start in a ‘radio’ mode.  This mode is implemented differently by various music providers, however generally it is a way of starting music playing of a given Artist/Song/Genre and then the player continues to select tracks similar to the given source (rather than loading a list of tracks into the media player queue).   I would personally use this to implement automations to play certain genre mixes of music at various times of the day, or even to allow playing of a random genre or artist mix with the press of a button.
 
The HA media_player currently supports two parameters related to starting music playing: ‘path’ and a ‘media type’.
 
Music providers may support the method of starting a ‘radio mode’ in various implementation-specific ways.  For example, the Roon music service supports this in their API call for ‘play_media’ by passing not only a ‘path’ parameter, but also an ‘action’ parameter.  This ‘action’ parameter specifies the ‘play action’ and can either be passed as: Null (which uses a default play action); ‘Play Now’; ‘Queue’ or ‘Start Radio’.  Using this parameter is the only way supported by the RoonAPI to start a Radio mode (the Roon API does support specifying a play action through the ‘path’ parameter, however this is limited to a play action such as ‘play artist’ or ‘play genre’ but does not support specifying the ‘Start Radio’ action).
 
The HA Roon Integration currently passes the HA media_player ‘path’ parameter to the RoonAPI ‘play_media’ function without specifying the ‘action’ parameter.
 
Options to implement this within HA could be:
 
Option 1:
Reuse the HA ‘media type’ parameter to pass to the Roon API as the ‘action’ parameter in the Roon API ‘play_media’ function call.  Thus a user could specify ‘Start Radio’ as the ‘media type’, and the HA Roon integration media_player code would be changed to pass this value to the Roon API call as the ‘action’ parameter. Advantages of this are that it is using an existing parameter and is simple to implement.  Disadvantages are that it only implements this for Roon media players (and not other music providers), and that it is a bit of a ‘hack’ as the ‘media_type’ parameter is not intended to be used as a play action.  In addition, it would not be obvious to users how to utilise this feature.

Option 2:
Delimit the action at the start or end of the path parameter, potentially enhancing the pyroon split_path function to do this. Advantages of this are that it is using an existing parameter and is simple to implement.  Disadvantages are that it would not be obvious to users how to utilise this feature.
 
Option 3:
Create a new HA service for the HA Roon Integration to perform a request to start a ‘Roon radio’.  Advantages are that it creates a distinct service for this Roon specific functionality obvious to users and is relatively simple to implement.  Disadvantages are that it only implements this for Roon media players (and not other music providers), thus services would need to be created uniquely for each music provider’s HA Integration.
 
Option 4:
Use a URL format like “roon-radio://……” that the AH Play Media API can process to make an appropriate call to the Roon API.  Items could be exposed via the media browser so users don’t actually have to craft the URLs themselves.  I am not familiar with this approach, so would appreciate further collaboration on the pros/cons of this option.
 
Option 5:
Add an additional argument to the HA Play_Media service to directly support the ‘play action’ request (aligning with the way the RoonAPI uses a specific parameter for this).  Advantages are that it is obvious to users and would align the implementation of this capability across all media player implementations.  Disadvantages are that it may not be supported by all music providers and may be more complex to implement as it impacts all media player integrations (and not just the HA Roon Integration).

Reply:
I'd change the text a bit as this is a generic proposal for all media players, not specific to Roon.

Option 6: Add Radio to the Enqueue enum, which is an existing option to the play_media call.

Reply:
thanks, I've made more generic and reordered the options (btw, I'd appreciate advice on how/where to raise this once were happy with the content): 

The purpose for this architecture discussion is to collaboratively identify the best method to enhance the HA media_player's play_media service to support requesting music to start in a ‘radio’ mode. This mode is implemented differently by various music player providers, however generally it is a way of starting to play music of a given Artist/Genre and then the player continuing to select tracks similar to the given source (rather than initially loading a list of tracks into the media player queue).

The HA media_player's play_media service currently supports the following parameters related to starting music playing:
* 'media_content_id' - The ‘path’/ID of the content to play. Platform dependent.
* ‘media_content_type’ - They type of content to play.
* 'enqueue' - If the content should be played now or be added to the queue (Play now/Play next/Add to queue/Play now and clear queue).
* 'announce' - If the media should be played as an announcement (true/false).

Music providers may support the method of starting a ‘radio mode’ in various implementation-specific ways. For example, the Roon music service supports this in their API call for ‘play_media’ by passing an ‘action’ parameter to initiate this mode. This ‘action’ parameter specifies the ‘play action’ and can either be passed as: Null (which uses a default play action); ‘Play Now’; ‘Queue’ or ‘Start Radio’.

Options to implement this within HA could be:

Option 1:
Add 'Play Radio' to the 'Enqueue' enumeration field as an option.  Advantages are that this is an existing field in the play_media call.

Option 2:
Add an additional field to the HA Play_Media service to directly request to 'start radio'.

Option 3:
Delimit the action at the start or end of the 'media_content_id'/'path' parameter, potentially enhancing the pyroon split_path function to do this. Advantages of this are that it is using an existing parameter. Disadvantages are that it would not be as obvious to users how to utilise this feature.

Option 4:
Create a new HA service for each provider's media player integration to perform a request to start a 'radio’. Advantages are that it creates a distinct service for this specific functionality. Disadvantages are that individual services would need to be created uniquely for each music provider’s HA Integration and therfore not be standardised across HA media players.

Option 5:
Use a URL format like “roon-radio://……” that the AH Play Media API can process to make an appropriate call to the Roon API. Items could be exposed via the media browser so users don’t actually have to craft the URLs themselves.

Thanks again for the support!